mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/transformation/mpsToModelix/initial/ModelSynchronizer.kt

```diff
@@ -123,83 +155,109 @@ class ModelSynchronizer(private val branch: IBranch, postponeReferenceResolution
+                        -1,
+                        BuiltinLanguages.MPSRepositoryConcepts.SingleLanguageDependency,
+                    )
+
+                nodeMap.put(model, languageModuleReference, cloudLanguageDependency.nodeIdAsLong())
+
+                // warning: might be fragile, because we synchronize the properties by hand
+                cloudLanguageDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.LanguageDependency.name,
+                    targetLanguageName,
                 )
 
-            nodeMap.put(model, languageModuleReference, cloudLanguageDependency.nodeIdAsLong())
-
-            // warning: might be fragile, because we synchronize the properties by hand
-            cloudLanguageDependency.setPropertyValue(
-                BuiltinLanguages.MPSRepositoryConcepts.LanguageDependency.name,
-                languageModuleReference?.moduleName,
-            )
-
-            cloudLanguageDependency.setPropertyValue(
-                BuiltinLanguages.MPSRepositoryConcepts.LanguageDependency.uuid,
-                languageModuleReference?.moduleId.toString(),
-            )
+                cloudLanguageDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.LanguageDependency.uuid,
+                    targetLanguageId,
+                )
 
-            cloudLanguageDependency.setPropertyValue(
-                BuiltinLanguages.MPSRepositoryConcepts.SingleLanguageDependency.version,
-                model.module.getUsedLanguageVersion(language).toString(),
-            )
+                cloudLanguageDependency.setPropertyValue(
+                    BuiltinLanguages.MPSRepositoryConcepts.SingleLanguageDependency.version,
+                    model.module.getUsedLanguageVersion(language).toString(),
+                )
+            }
         }
 
     fun addDevKitDependency(model: SModel, devKit: SModuleReference) =
         syncQueue.enqueue(linkedSetOf(SyncLock.MODELIX_WRITE, SyncLock.MPS_READ), SyncDirection.MPS_TO_MODELIX) {
             val modelixId = nodeMap[model]!!
+            val cloudNode = branch.getNode(modelixId)
+            val childLink = BuiltinLanguages.MPSRepositoryConcepts.Model.usedLanguages
 
             val repository = model.repository
             val devKitModuleId = devKit.moduleId
             val devKitModule = repository.getModule(devKitModuleId)
+            val devKitName = devKitModule?.moduleName
+            val devKitId = devKitModule?.moduleId.toString()
+
+            // duplicate check and sync
+            val dependencyExists = cloudNode.getChildren(childLink)
+                .firstOrNull { devKitId == it.getPropertyValue(BuiltinLanguages.MPSRepositoryConcepts.LanguageDependency.uuid) } != null
+            if (dependencyExists) {
+                logger.warn { "Model ${model.name}'s DevKit Dependency for $devKitName will not be synchronized, because it already exists on the server." }
```

Comment:
> Warn user if they are present and ask if he/she wants to remove them from the project automatically (that's the recommended way, unrecommended is to keep them but then it might break the model/project's consistency).

See https://issues.modelix.org/issue/MODELIX-755

Will the users even see this warning? I guess we let the sync continue silently with incomplete data.
(This holds basically for all checks.)